dcos/http/client/client.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/dcos/dcos-go/jwt/transport"
+)
+
+// loadCAPool will load a valid x509 cert.
+func loadCAPool(path string) (*x509.CertPool, error) {
+	caPool := x509.NewCertPool()
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	if !caPool.AppendCertsFromPEM(b) {
+		return nil, errors.New("CACertFile parsing failed")
+	}
+
+	return caPool, nil
+}
+
+// getTransport will return transport for http.Client
+func getTransport(caCertificatePath string) (*http.Transport, error) {
+	tr := &http.Transport{}
+	// if user provided CA cert we must use it, otherwise use InsecureSkipVerify: true for all HTTPS requests.
+	if caCertificatePath != "" {
+		caPool, err := loadCAPool(caCertificatePath)
+		if err != nil {
+			return tr, err
+		}
+
+		tr.TLSClientConfig = &tls.Config{
+			RootCAs: caPool,
+		}
+	} else {
+		tr.TLSClientConfig = &tls.Config{
+			InsecureSkipVerify: true,
+		}
+	}
+	return tr, nil
+}
+
+// NewDCOSClient returns a client with CAPool when passed a CA certificate path and
+// a Authorization header when passed a IAM configuration path. If the path to CA
+// certificate or IAM config is an empty string then the corrosponding client
+// configurations which they set are ignored.
+func NewDCOSClient(caCertificatePath string, iamConfigPath string) (*http.Client, error) {
```

Comment:
maybe instead of returning `*http.Client` we should return `http.RoundTripper`?
so we can make a roundtripper decorator similar to https://github.com/dcos/dcos-go/blob/master/jwt/transport/transport.go#L48 and implement the IAM Config as a functional parameter. We could also refactor `jwt/transport` to include this functionality with a new option? So we could handle JWT and CA certs all from one place.

i feel like `http.Client` constructor is too broad.
@jdef what's your opinion?